### PR TITLE
Saas 18 sale stock mrp logged notes vas

### PR DESCRIPTION
**Pull request by vas-odoo**
TASK:- https://www.odoo.com/web?#id=36098&view_type=form&model=project.task&action=333&active_id=131&menu_id=4720
PAD:- https://pad.odoo.com/p/r.8e4cf8dd80b799b1adabb4bfb32001e3

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mrp/models/mrp_production.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def _get_default_location_dest_id(self):
     product_id = fields.Many2one(
         'product.product', 'Product',
         domain=[('type', 'in', ['product', 'consu'])],
-        readonly=True, required=True,
+        readonly=True, required=True, track_visibility='onchange',
```
Comment (jco-odoo) [inline addons/mrp/models/mrp_production.py]: on product_qty then?

File: addons/mrp/models/stock_move.py [hunk 2]
```diff
@@ -76,6 +81,14 @@ def _compute_is_done(self):
         for move in self:
             move.is_done = (move.state in ('done', 'cancel'))
 
+    @api.multi
+    def write(self, vals):
+        if vals.get('product_uom_qty'):
+            for move in self.filtered(lambda x : x.state not in ('done', 'draft')):
+                production = move.production_id or move.raw_material_production_id
+                move.log_message(production, 'mrp.track_production_move_template', vals, {})
+        return super(StockMove, self).write(vals)
```
Comment (jco-odoo) [inline addons/mrp/models/stock_move.py]: This should only be for pickings normally. (e.g. I don't want plenty of messages when you update the qty on a production order)

File: addons/mrp/models/stock_move.py [hunk 3]
```diff
@@ -31,11 +31,16 @@ def _check_lot_id(self):
...
+        data = {}
+        if vals.get('location_id'):
+            data['location_name'] = self.env['stock.location'].browse(vals['location_id']).name
+        for movelot in self.filtered(lambda x: x.state == 'done'):
+            if 'lot_id' in vals:
+                data['lot_name'] = self.env['stock.production.lot'].browse(vals['lot_id']).name
                 movelot.move_id.production_id.move_raw_ids.mapped('move_line_ids')\
                     .filtered(lambda r: r.done_wo and not r.done_move and r.lot_produced_id == movelot.lot_id)\
                     .write({'lot_produced_id': vals['lot_id']})
+            movelot.log_message(movelot.production_id, 'mrp.track_production_move_template', vals, data)
```
Comment (jco-odoo) [inline addons/mrp/models/stock_move.py]: location_id / owner_id/.... should only be within the done movelots for loop

**Comment by amoyaux** [comment]
Pushed on saas-18
513c1474a395f50918fb88b75b6627bc44da80d6
41a3c884c07738348cea7d38d8bc79ba6ee6ad7a
b509427e7a8683c00a0b977d19b0a473bad0d82c
0c8e18eba6aeb8d722995b528f2a647933871b79
a6114ac371736ea8f51ac174c022db24da110288

**Pull request closed without merging**